.grit/patterns/go/exported_loop_pointer.md

```diff
@@ -0,0 +1,104 @@
+---
+title: Fix pointer is shared between loop iterations
+---
+
+`$VALUE` serves as a loop pointer that might be exported from the loop. Since this pointer is shared across loop iterations, the exported reference will consistently point to the last loop value, potentially leading to unintended consequences. To address this issue, duplicate the pointer within the loop to ensure each iteration has its own distinct reference.
+
+### references
+
+- [looppointer](https://github.com/kyoh86/looppointer)
+
+tags: #fix #correctness
+
+```grit
+language go
+
+`for _, $val := range $values { $body }` where {
+    $body <: not contains `$val := $val`,
+    $body <: contains `&$val`
+} => `for _, $val := range $values { 
+        $val := $val \n $body 
+    }`
+```
+
+## loop iterations with pointers
```

Comment:
In all of your tests, please stop combining multiple cases into one code block. Make a separate header and before/after pair for each case. This should be 4 separate cases.

Reply:
yes true, I will keep this in mind, updated it now.